src/credentials/FabricTable.cpp

```diff
@@ -182,31 +183,26 @@ CHIP_ERROR FabricInfo::LoadFromStorage(PersistentStorageDelegate * storage)
     return err;
 }
 
-CHIP_ERROR FabricInfo::GeneratePeerId(FabricId fabricId, NodeId nodeId, PeerId * compressedPeerId) const
+CHIP_ERROR FabricInfo::ComputeCompressedFabricId(FabricId fabricId, NodeId nodeId, const P256PublicKeySpan & rootPubkeySpan,
+                                                 CompressedFabricId & result)
 {
-    ReturnErrorCodeIf(compressedPeerId == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    uint8_t compressedFabricIdBuf[sizeof(uint64_t)];
-    MutableByteSpan compressedFabricIdSpan(compressedFabricIdBuf);
-    P256PublicKey rootPubkey;
-
-    {
-        P256PublicKeySpan rootPubkeySpan;
-        ReturnErrorOnFailure(GetRootPubkey(rootPubkeySpan));
-        rootPubkey = rootPubkeySpan;
-    }
-
     ChipLogDetail(Inet, "Generating compressed fabric ID using uncompressed fabric ID 0x" ChipLogFormatX64 " and root pubkey",
                   ChipLogValueX64(fabricId));
-    ChipLogByteSpan(Inet, ByteSpan(rootPubkey.ConstBytes(), rootPubkey.Length()));
-    ReturnErrorOnFailure(GenerateCompressedFabricId(rootPubkey, fabricId, compressedFabricIdSpan));
-    ChipLogDetail(Inet, "Generated compressed fabric ID");
-    ChipLogByteSpan(Inet, compressedFabricIdSpan);
-
-    // Decode compressed fabric ID accounting for endianness, as GenerateCompressedFabricId()
-    // returns a binary buffer and is agnostic of usage of the output as an integer type.
-    CompressedFabricId compressedFabricId = Encoding::BigEndian::Get64(compressedFabricIdBuf);
-    compressedPeerId->SetCompressedFabricId(compressedFabricId);
-    compressedPeerId->SetNodeId(nodeId);
+    ChipLogByteSpan(Inet, rootPubkeySpan);
+    ReturnErrorOnFailure(GenerateCompressedFabricId(P256PublicKey(rootPubkeySpan), fabricId, result));
+    ChipLogDetail(Inet, "Generated compressed fabric ID 0x" ChipLogFormatX64, ChipLogValueX64(result));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR FabricInfo::BuildNewFabric(FabricId fabricId, NodeId nodeId, const chip::ByteSpan & rootCert, FabricInfo & result)
+{
+    result.mNodeId = nodeId;
+    result.SetRootCert(rootCert);
```

Comment:
This ties the root cert to the FabricInfo which we are trying to remove right now. Please sync with Boris